src/lib/rest/StringFilter.h

```diff
@@ -0,0 +1,159 @@
+*   op                   the operator of the filter-item
+*   valueType            the type of the right-hand-side
+*   numberValue          if 'valueType' is 'numeric', the value is stored here
+*   stringValue          if 'valueType' is 'string', the value isstored here
+*   boolValue            if 'valueType' is 'boolean', the value is stored here
+*   stringList           vector/list of string values
+*   numberList           vector/list of numeric values
+*   numberRangeFrom      lower limit for numeric ranges
+*   numberRangeTo        upper limit for numeric ranges
+*   stringRangeFrom      lower limit for string ranges
+*   stringRangeTo        upper limit for string ranges
+*   attributeName        The name of the attribute, used for unary operators only
+*   
+*/
+class StringFilterItem
+{
+public:
+  std::string               left;
+  StringFilterOp            op;
+  StringFilterValueType     valueType;
+  double                    numberValue;
+  std::string               stringValue;
+  bool                      boolValue;
+  std::vector<std::string>  stringList;
+  std::vector<double>       numberList;
+  double                    numberRangeFrom;
+  double                    numberRangeTo;
+  std::string               stringRangeFrom;
+  std::string               stringRangeTo;
+  std::string               attributeName;  // Used for unary operators only
+
+  bool                      parse(char* qItem, std::string* errorStringP);
+  const char*               opName(void);
+  const char*               valueTypeName(void);
+
+  bool                      valueGet(char*                   s,
+                                     StringFilterValueType*  valueTypeP,
+                                     double*                 doubleP,
+                                     std::string*            stringP,
+                                     bool*                   boolP,
+                                     std::string* errorStringP);
+
+  bool                      valueParse(char* s, std::string* errorStringP);
+  bool                      rangeParse(char* s, std::string* errorStringP);
+  bool                      listParse(char* s, std::string* errorStringP);
+  bool                      listItemAdd(char* s, std::string* errorStringP);
+  bool                      matchEquals(ContextAttribute* caP);
+  bool                      matchGreaterThan(ContextAttribute* caP);
+  bool                      matchLessThan(ContextAttribute* caP);
+};
+
+
+
+class ContextElementResponse;
+/* ****************************************************************************
+*
+* StringFilter - 
+*
+* FIELDS
+*   filters       vector of filter-items, see StringFilterItem comment header
+*   mongoFilters  the filter-items translated to filters that can be used by mongo
+*                 Note that the method 'mongoFilterPopulate' must be called to translate
+*                 'filters' to 'mongoFilters'
```

Comment:
Good comment. However, what about using a similar one for match(), e.g:

"Note that the method 'match' needs to be called after 'parse'"

(Or maybe I'm wrong and it is not really needed... :)